ui/src/app/store/zone/types/base.ts

```diff
@@ -12,4 +15,41 @@ export type Zone = Model & {
   updated: string;
 };
 
-export type ZoneState = GenericState<Zone, APIError>;
+export type ZonePK = Zone[typeof ZONE_PK];
+
+export type ZoneProcess = {
+  processing: ZonePK[];
+  success: ZonePK[];
+};
+
+export type ZoneProcesses = {
+  [ZONE_ACTIONS.delete]: ZoneProcess;
+  [ZONE_ACTIONS.update]: ZoneProcess;
+};
+
+export type ZoneAPIMeta = {
+  formId: string | null;
+  modelPk: ZonePK | null;
+};
+
+export type ZoneAPISuccess<R> = {
+  result: R;
+};
+
+export type ZoneAPIError = ZoneAPIMeta & {
+  error: APIError;
+};
+
+export type ZoneStateError = ZoneAPIError & {
+  action: ValueOf<typeof ZONE_ACTIONS>;
+};
```

Comment:
The new error shape in state includes `action` (the action that errored), `modelPk` (the primary key of the model the action was done to, if applicable) and `formId` (the id of the form from which the action was dispatched).